### PR TITLE
fix: ACNA-2219 - add OAuth Server-to-Server api calls and e2e tests

### PR DESCRIPTION
**Pull request by shazron**
## How Has This Been Tested?

- npm run e2e (all tests pass)
- npm test (all tests pass with 100% code coverage)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: bin/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "rules": {
+      "node/no-unpublished-require": 0
+  }
+}
```
Comment (amulyakashyap09) [inline bin/.eslintrc.json:5]: We can remove the last empty line.

**Review by amulyakashyap09 — APPROVED** [review]
LGTM!

File: bin/.eslintrc.json (hunk 1 above)
Comment (shazron) [reply bin/.eslintrc.json:5]: > We can remove the last empty line.

there is no empty last line. The red circle with a red dash in it signifies no newline at the end of the file. You can verify this on disk.

**Review by moritzraho — APPROVED** [review]
lgtm!
just curious why the patch was needed

**Review by MichaelGoberling — APPROVED** [review]

**Pull request merged**